ocsp.go

```diff
@@ -512,18 +512,18 @@ func isValidOCSPStatus(status ocspStatusCode) bool {
 // verifyPeerCertificate verifies all of certificate revocation status
 func verifyPeerCertificate(verifiedChains [][]*x509.Certificate) (err error) {
 	for i := 0; i < len(verifiedChains); i++ {
-		n := len(verifiedChains[i]) - 1
-		if !verifiedChains[i][n].IsCA || string(verifiedChains[i][n].RawIssuer) != string(verifiedChains[i][n].RawSubject) {
+		numberOfNoneRootCA := len(verifiedChains[i]) - 1
```

Comment:
This part of the code was a little confusing to me at first so I'd suggest to put some comments in here. Something like - "Certificate signed by Root CA. This should be one before the last in the Certificate Chain".